### PR TITLE
Use a per-mode-ciphersuite DST for deterministic key derivation

### PR DESCRIPTION
**Pull request by armfazh**
no updates in the spec, this updates the code.

Adds the version as prefix in several parts.

### CODE REVIEW

**Comment by chris-wood** [comment]
#243 changed the domain separation strings such that they're of the following general form:

```
LABEL-STRING || VERSION || MODE || SUITE-ID
```

So, for example, the finalize DST is:

```
"Finalize-" || "VOPRF06-" || MODE || SUITE-ID
```

This change seems to make the code do something like:

```
"VOPRF06-" || "Finalize-" || MODE || SUITE-ID
```

Which is a spec change. Was that intended? (For context, the change in #243 was to allow better re-use of internal functions for the DLEQ proofs _outside_ of this spec.)

**Comment by armfazh** [comment]
> chris-wood:
> #243 changed the domain separation strings such that they're of the following general form:
> Which is a spec change. Was that intended? 

No, I did changes in a old version. I fix that now.
The only "breaking change" is on `DeriveKeyPair`, now it uses the DST dervied per-mode and per-ciphersuite. (This was commented in #236 )

File: poc/oprf.sage [hunk 1]
```diff
@@ -289,31 +293,29 @@ class VerifiableServerContext(ServerContext,Verifiable):
 
 mode_base = 0x00
```
Comment (chris-wood) [inline poc/oprf.sage:294]: Unrelated, but should we also make these constant names all-caps?

**Review by chris-wood — APPROVED** [review]

**Comment by chris-wood** [comment]
@armfazh can we file an issue to track updating this in CIRCL?

File: poc/oprf.sage (hunk 1 above)
Comment (armfazh) [reply poc/oprf.sage]: done

**Pull request merged**